### ISSUE TITLE
Windows, getting started, running both the server and client on localhost, without Docker

### ISSUE THREAD

**Issue by kurtcodemander** [antoniomika/sish#234]
Hi, this is a great project, but I'm having some problems getting started on Windows. 

I have downloaded Sish for Windows and I want to try it on my development machine as simply as possible. 

For testing I want to forward HTTP port 80 => HTTP port 5000. 

I have tried disabling my firewall and also tried many other command line arguments other than those shown below, but always the same error message. 

```
sish --authentication=false
2022/06/24 - 05:32:41 | Starting SSH service on address: localhost:2222
2022/06/24 - 05:32:41 | Loading ssh_key as ssh-ed25519 host key
```

`ssh -p 2222 -R 80:localhost:5000 kurti@localhost`

Then the output from the ssh client is: 

```
Press Ctrl-C to close the session.

Warning: remote port forwarding failed for listen port 80
```

I can see that port 80 has been opened by sish and is listening to localhost/127.0.0.1/0.0.0.0. 

Does this have something to do with a missing entry for localhost:80 or localhost:5000 in known_hosts? 

Here is the debug output from sish: 
```
[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
 - using env:   export GIN_MODE=release
 - using code:  gin.SetMode(gin.ReleaseMode)

2022/06/24 - 05:44:06 | Starting SSH service on address: localhost:2222
[GIN-debug] Loaded HTML Templates (9):
        - console
        - footer
        - header
        - routes.tmpl
        -
        - footer.tmpl
        - header.tmpl
        - routes
        - console.tmpl

2022/06/24 - 05:44:06 | =======Start=========
2022/06/24 - 05:44:06 | ===Goroutines=====
2022/06/24 - 05:44:06 | 12
2022/06/24 - 05:44:06 | ===Listeners======
2022/06/24 - 05:44:06 | ===Clients========
2022/06/24 - 05:44:06 | ===HTTP Clients===
2022/06/24 - 05:44:06 | ===TCP Aliases====
2022/06/24 - 05:44:06 | ===Web Console Routes====
2022/06/24 - 05:44:06 | ===Web Console Tokens====
2022/06/24 - 05:44:06 | ========End==========
2022/06/24 - 05:44:06 | Loading ssh_key as ssh-ed25519 host key
2022/06/24 - 05:44:08 | =======Start=========
2022/06/24 - 05:44:08 | ===Goroutines=====
2022/06/24 - 05:44:08 | 11
2022/06/24 - 05:44:08 | ===Listeners======
2022/06/24 - 05:44:08 | localhost:2222 &{0xc000394780 {<nil> 0}}
2022/06/24 - 05:44:08 | ===Clients========
2022/06/24 - 05:44:08 | ===HTTP Clients===
2022/06/24 - 05:44:08 | ===TCP Aliases====
2022/06/24 - 05:44:08 | ===Web Console Routes====
2022/06/24 - 05:44:08 | ===Web Console Tokens====
2022/06/24 - 05:44:08 | ========End==========
2022/06/24 - 05:44:10 | =======Start=========
2022/06/24 - 05:44:10 | ===Goroutines=====
2022/06/24 - 05:44:10 | 11
2022/06/24 - 05:44:10 | ===Listeners======
2022/06/24 - 05:44:10 | localhost:2222 &{0xc000394780 {<nil> 0}}
2022/06/24 - 05:44:10 | ===Clients========
2022/06/24 - 05:44:10 | ===HTTP Clients===
2022/06/24 - 05:44:10 | ===TCP Aliases====
2022/06/24 - 05:44:10 | ===Web Console Routes====
2022/06/24 - 05:44:10 | ===Web Console Tokens====
2022/06/24 - 05:44:10 | ========End==========
2022/06/24 - 05:44:12 | =======Start=========
2022/06/24 - 05:44:12 | ===Goroutines=====
2022/06/24 - 05:44:12 | 11
2022/06/24 - 05:44:12 | ===Listeners======
2022/06/24 - 05:44:12 | localhost:2222 &{0xc000394780 {<nil> 0}}
2022/06/24 - 05:44:12 | ===Clients========
2022/06/24 - 05:44:12 | ===HTTP Clients===
2022/06/24 - 05:44:12 | ===TCP Aliases====
2022/06/24 - 05:44:12 | ===Web Console Routes====
2022/06/24 - 05:44:12 | ===Web Console Tokens====
2022/06/24 - 05:44:12 | ========End==========
2022/06/24 - 05:44:12 | Accepted SSH connection for: 127.0.0.1:61187
2022/06/24 - 05:44:13 | Main Request Info tcpip-forward true            localhost   P
2022/06/24 - 05:44:13 | Main Channel Info session
2022/06/24 - 05:44:13 | Handling session for connection: &{session [] 0 0 32768 16384 0xc0004157a0 true 0 0xc0004804e0 {0 0} 0xc000480480 false {0xc00041e340 1048576 0 false} 0xc00059a080 0xc00059a100 {0 0} 2097152 {0 0} false map[]}
2022/06/24 - 05:44:13 | Sub Channel Type pty-req true    ♫xterm-256color   �   C  ☻�  ☺�   ♂�  %��  %�
2022/06/24 - 05:44:14 | =======Start=========
2022/06/24 - 05:44:14 | ===Goroutines=====
2022/06/24 - 05:44:14 | 23
2022/06/24 - 05:44:14 | ===Listeners======
2022/06/24 - 05:44:14 | localhost:2222 &{0xc000394780 {<nil> 0}}
2022/06/24 - 05:44:14 | ===Clients========
2022/06/24 - 05:44:14 | 127.0.0.1:61187 &{0xc000004cd8 0xc000488110 0xc0003f93e0 0xc0005450e0 0xc000545140 0xc0005451a0 0  true false  false false false 0xc000545200 true 0xc0003f93d8}
2022/06/24 - 05:44:14 | ===HTTP Clients===
2022/06/24 - 05:44:14 | ===TCP Aliases====
2022/06/24 - 05:44:14 | ===Web Console Routes====
2022/06/24 - 05:44:14 | ===Web Console Tokens====
2022/06/24 - 05:44:14 | ========End==========
2022/06/24 - 05:44:15 | Aborting with error cannot find connection for host: localhost
2022/06/24 - 05:44:15 | localhost |←[90;43m 404 ←[0m|         515µs |       127.0.0.1 |←[97;44m GET     ←[0m /
Error #01: cannot find connection for host: localhost
2022/06/24 - 05:44:16 | =======Start=========
2022/06/24 - 05:44:16 | ===Goroutines=====
2022/06/24 - 05:44:16 | 25
2022/06/24 - 05:44:16 | ===Listeners======
2022/06/24 - 05:44:16 | localhost:2222 &{0xc000394780 {<nil> 0}}
2022/06/24 - 05:44:16 | ===Clients========
2022/06/24 - 05:44:16 | 127.0.0.1:61187 &{0xc000004cd8 0xc000488110 0xc0003f93e0 0xc0005450e0 0xc000545140 0xc0005451a0 0  true false  false false false 0xc000545200 true 0xc0003f93d8}
2022/06/24 - 05:44:16 | ===HTTP Clients===
2022/06/24 - 05:44:16 | ===TCP Aliases====
2022/06/24 - 05:44:16 | ===Web Console Routes====
2022/06/24 - 05:44:16 | ===Web Console Tokens====
2022/06/24 - 05:44:16 | ========End==========

```

**Comment by antoniomika** [comment]
Hey @kurtcodemander!

Looks like you're most of the way there!

sish operates (mainly) in two modes, HTTP(S) vhosting mode and TCP proxying mode. This allows you to deploy hosta.example.com alongside hostb.example.com and sish will know which ssh client to send the traffic to.

TCP proxying means port 80 on the sish server will be forwarded to a specific ssh client, meaning you can't have multiple websites operate on port 80 or 443. Which of the two are you trying to do?

If doing the first option (HTTP(S) vhosting), sish also goes a bit further to reserve some domains to prevent a user from binding them. Mainly `localhost` as that has a few meanings in ssh tunnels, and is sent when no host is provided. You can set sish to randomly generate a hostname as well. Just note that your hosts entry will need to include this domain so sish knows the hostname for the request. Otherwise, you can allow any domain to be bound and access it over localhost.

There was an issue with listeners being setup on windows though. That's fixed now with v2.4.0, so give that a try and let me know!

**Comment by kurtcodemander** [comment]
Thanks, I'm now using 2.4.0 and I've got a better grasp of how sish works so it's working for me now.
